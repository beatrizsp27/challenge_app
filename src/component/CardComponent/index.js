import React from 'react';

const CardComponent = () => {
	return <h1>Hola mundo MainDetailsProduct </h1>;
};

export default CardComponent;
