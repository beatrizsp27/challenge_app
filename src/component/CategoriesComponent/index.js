import React from "react";

const CategoriesComponent = ({categoriesArray}) =>{
    return(
        <div className={'screens_component_category'}>
            {
                categoriesArray && categoriesArray.length > 0 && (
                    <>
                        <div  className={'screens_category'}>
                            <div  className={'list_ul'}>
                                {categoriesArray.map((categoryItem, index)=> {
                                    const categoriesArrayLength = categoriesArray.length;
                                    const isOtherFormat = index+1 === categoriesArrayLength;
                                    const sign = ' >'
                                    return (
                                        // eslint-disable-next-line react/jsx-key
                                        <li key={index} >
                                            <span  className={isOtherFormat? 'screens_title_body_header_bold' :'screens_title_body_header'}> {isOtherFormat ? categoryItem : `${categoryItem}${sign}`} </span>
                                        </li>
                                    )})}
                            </div>
                        </div>
                    </>
                )
            }
        </div>
    );
};

export default CategoriesComponent;
