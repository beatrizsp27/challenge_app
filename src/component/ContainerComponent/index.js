import React from 'react';

const ContainerComponent = ({ children }) => {
    return <div className='screens_component_div'>{children}</div>;
};

export default ContainerComponent;
