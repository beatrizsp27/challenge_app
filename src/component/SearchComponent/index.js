import React from 'react';
import {logo, search} from "../../assets";
import s from "../../strings";

const SearchComponent = ({ handleChange, onAction }) => {
    return (
        <div className={'screens_details_header'}>
            <form onSubmit={onAction}>
                <div className={'screens_details_form'}>
                    <div className={'box_img_search'}>
                        <img className={'box_img'} src={logo}/>
                    </div>
                    <div className={'box_input_icon_search'} >
                        <input
                            autoComplete={'false'}
                            name={'search'}
                            className="box_input_text"
                            type="search"
                            placeholder={s.search}
                            onChange={handleChange}
                        />
                        <div className={'box_search_icon'} onClick={onAction}>
                            <img  className={'box_img_mel'}  src={search} alt={''}/>
                        </div>
                    </div>
              </div>
            </form>
        </div>
    )
};

export default SearchComponent;
