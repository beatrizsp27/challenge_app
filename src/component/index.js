export { default as ContainerComponent } from './ContainerComponent';
export { default as SearchComponent } from './SearchComponent';
