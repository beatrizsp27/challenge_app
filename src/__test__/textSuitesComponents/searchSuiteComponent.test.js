import React from "react";
import { SearchComponent } from '../../component';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

/** SE CONFIGURA EL ADAPTADOR**/
import {shallow, configure} from "enzyme";
configure({ adapter: new Adapter() })

describe("ContainerComponent", ()=>{

    /** VARIABLES**/
    let wrapperSearch;
    const handleChange = jest.fn();
    const prevent = jest.fn();
    const onAction = jest.fn();

    beforeAll(()=>{
        /** SE MONTA EL COMPONENTE**/
        wrapperSearch = shallow(<SearchComponent handleChange={handleChange} onAction={onAction}/>);
    });

    /** SE VALIDA QUE EL COMPONENTE ESTE DEFENIDO Y QUE EXISTA**/
    test("renderear correctamente", () => {
        expect(wrapperSearch).toBeTruthy()
        wrapperSearch.find('img');
        expect(wrapperSearch.length).toEqual(1);
    })

    /** SE VALIDA QUE EL FORMULARIO FUNCIONE CORRECTAMENTE**/
    test("llamar al form enviando datos y validando que fueron recibidos ยก", () => {
        wrapperSearch.find('input').simulate('change', { target : {values: 'Esta es una prueba'}});
        wrapperSearch.find('form').simulate('submit', {preventDefault: prevent});
        expect(handleChange.mock.calls).toEqual([
            [
                {
                    "target": {
                        "values": "Esta es una prueba"
                    }
                }
            ]
        ]);
        expect(prevent.mock.calls).toEqual([]);
    });

    /** SE VALIDA QUE EL FORMULARIO FUNCIONE CORRECTAMENTE**/
    test("llamar al form enviando datos y validando que fueron recibidos y probando onClick", () => {
        /** ENVIO DE DATOS**/
        wrapperSearch.find('input').simulate('change', { target : {values: 'Esta es una prueba 2'}});
        wrapperSearch.find('form').simulate('submit', {preventDefault: prevent});

        /** SE VALDIA QUE EXISTA EL NODO DONDE SE ESNCUNTRA EL CLIC**/
        expect(wrapperSearch.find('div')).toHaveLength(3);

        /** SE SIMULA EL CLIC DE LA BUSQUEDA**/
        const simulateOnCLick = wrapperSearch.find('div').at(2);
        simulateOnCLick.simulate('click');
        expect(handleChange.mock.calls).toEqual([
            [
                {
                    "target": {
                        "values": "Esta es una prueba 2"
                    }
                }
            ]
        ]);
        /** SE VALIDA QUE SE HAYA EJECUTADO SOLAMENTE UNA VEZ**/
        expect(prevent.mock.calls).toEqual([]);
    });

    /** SE GENERA EL SNAPSHAP PARA VALIDAR LOS QUE LOS CAMBIOS DEL COMPONENTE SEAN CONTROLADOS**/
    test("Se crea el snap despues de las pruebas", () => {
        const treeSearch = renderer.create(<SearchComponent />).toJSON();
        expect(treeSearch).toMatchSnapshot();
    })
});
