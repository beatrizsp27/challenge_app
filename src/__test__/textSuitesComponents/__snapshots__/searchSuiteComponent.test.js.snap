// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`* ContainerComponent Se crea el snap 1`] = `
<div
  className="screens_details_header"
>
  <div>
    <img
      className="box_img_mel"
      src="Logo_ML@2x.png.png"
    />
  </div>
  <input
    className="box_input_text"
    placeholder="Nunca dejes de buscar"
    type="search"
  />
  <div
    className="box_search_icon"
  >
    <img
      alt=""
      className="box_img"
      src="ic_Search@2x.png.png"
    />
  </div>
</div>
`;

exports[`ContainerComponent Se crea el snap 1`] = `
<form>
  <div
    className="screens_details_header"
  >
    <div>
      <img
        className="box_img_mel"
        src="Logo_ML@2x.png.png"
      />
    </div>
    <input
      className="box_input_text"
      name="search"
      placeholder="Nunca dejes de buscar"
      type="search"
    />
    <div
      className="box_search_icon"
    >
      <img
        alt=""
        className="box_img_mel"
        src="ic_Search@2x.png.png"
      />
    </div>
  </div>
</form>
`;
